protocol/messenger.go

```diff
@@ -415,32 +422,42 @@ func (m *Messenger) processSentMessages(ids []string) error {
 		if err != nil {
 			return errors.Wrapf(err, "Can't save raw message marked as sent")
 		}
+
+		err = m.UpdateMessageOutgoingStatus(id, "sent")
+		if err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
-func shouldResendEmojiReaction(message *common.RawMessage, t common.TimeSource) (bool, error) {
-	if message.MessageType != protobuf.ApplicationMetadataMessage_EMOJI_REACTION {
-		return false, errors.New("Should resend only emoji reactions")
+func shouldResendMessage(message *common.RawMessage, t common.TimeSource) (bool, error) {
+	if !(message.MessageType == protobuf.ApplicationMetadataMessage_EMOJI_REACTION ||
+		message.MessageType == protobuf.ApplicationMetadataMessage_CHAT_MESSAGE) {
+		return false, errors.Errorf("Should resend only specific types of messages, can't resend %v", message.MessageType)
 	}
 
 	if message.Sent {
 		return false, errors.New("Should resend only non-sent messages")
 	}
 
-	if message.SendCount > emojiResendMaxCount {
+	if message.SendCount > messageResendMaxCount {
 		return false, nil
 	}
 
 	//exponential backoff depends on how many attempts to send message already made
-	backoff := uint64(math.Pow(2, float64(message.SendCount-1))) * emojiResendMinDelay * uint64(time.Second)
+	backoff := uint64(math.Pow(2, float64(message.SendCount-1))) * messageResendMinDelay * uint64(time.Second.Milliseconds())
```

Comment:
Why is the timing done in milliseconds and not nanoseconds? Just curious.

Reply:
Probably because `t.GetCurrentTime()` and `message.LastSent` are in milliseconds. Previously there was a mix of milliseconds and nanoseconds  because of which this code never worked as expected (the message would be resent only in ~350 days).